accessors/spanner/spanner_accessor.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package spanneraccessor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	"cloud.google.com/go/spanner/admin/database/apiv1/databasepb"
+	"cloud.google.com/go/spanner/admin/instance/apiv1/instancepb"
+	spanneradmin "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/clients/spanner/admin"
+	spannerclient "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/clients/spanner/client"
+	spinstanceadmin "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/clients/spanner/instanceadmin"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/utils"
+	"google.golang.org/api/iterator"
+)
+
+type SpannerAccessor interface {
+	GetDatabase(ctx context.Context, dbURI string) (*databasepb.Database, error)
+	GetDatabaseDialect(ctx context.Context, dbURI string) (string, error)
+	CheckExistingDb(ctx context.Context, dbURI string) (bool, error)
+	CreateEmptyDatabase(ctx context.Context, dbURI string) error
+	GetSpannerLeaderLocation(ctx context.Context, instanceURI string) (string, error)
+	CheckIfChangeStreamExists(ctx context.Context, changeStreamName, dbURI string) (bool, error)
+	ValidateChangeStreamOptions(ctx context.Context, changeStreamName, dbURI string) error
+	CreateChangeStream(ctx context.Context, changeStreamName, dbURI string) error
+}
+
+type SpannerAccessorImpl struct{}
+
+func (sp *SpannerAccessorImpl) GetDatabase(ctx context.Context, dbURI string) (*databasepb.Database, error) {
+	adminClient, err := spanneradmin.GetOrCreateClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return adminClient.GetDatabase(ctx, &databasepb.GetDatabaseRequest{Name: dbURI})
+}
+
+func (sp *SpannerAccessorImpl) GetDatabaseDialect(ctx context.Context, dbURI string) (string, error) {
+	result, err := sp.GetDatabase(ctx, dbURI)
+	if err != nil {
+		return "", fmt.Errorf("cannot connect to database: %v", err)
+	}
+	return strings.ToLower(result.DatabaseDialect.String()), nil
+}
+
+// CheckExistingDb checks whether the database with dbURI exists or not.
+// If API call doesn't respond then user is informed after every 5 minutes on command line.
+func (sp *SpannerAccessorImpl) CheckExistingDb(ctx context.Context, dbURI string) (bool, error) {
```

Comment:
I have just moved this function from an existing conversion function, added a tracking bug for this. Don't want to modify this further since this is called in multiple places.